core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -315,14 +326,23 @@ private[spark] class Executor(
       } catch {
         case ffe: FetchFailedException =>
           val reason = ffe.toTaskEndReason
+          setTaskFinished(true)
+          // Reset the interrupted status of the thread to update the status
+          Thread.interrupted()
```

Comment:
Could you move this into `setTaskFinished` and add comments to explain why we need to clear the interrupted status and add the jira number to the comments as well?